pallets/pools/src/lib.rs

```diff
@@ -1248,6 +1301,47 @@ pub mod pallet {
 			Ok(state)
 		}
 
+		pub(crate) fn do_update_pool(
+			pool_id: &T::PoolId,
+			changes: &PoolChanges<T::InterestRate>,
+		) -> DispatchResult {
+			Pool::<T>::try_mutate(pool_id, |pool| -> DispatchResult {
+				let pool = pool.as_mut().ok_or(Error::<T>::NoSuchPool)?;
+
+				if let Change::NewValue(min_epoch_time) = changes.min_epoch_time {
+					pool.parameters.min_epoch_time = min_epoch_time;
+				}
+
+				if let Change::NewValue(max_nav_age) = changes.max_nav_age {
+					pool.parameters.max_nav_age = max_nav_age;
+				}
+
+				if let Change::NewValue(tranches) = &changes.tranches {
+					let now = Self::now();
+
+					pool.tranches.combine_with_mut_non_residual_top(
```

Comment:
Aren't the tranches in the `Vec<TranchInput>` sorted from residual-to-non-residual? Then `combine_with_mut_residual_top`

Reply:
Yeah good point. I guess it was wrong in the old `update_tranches` implementation then, I copied it from there: https://github.com/centrifuge/centrifuge-chain/blob/parachain/pallets/pools/src/lib.rs#L651

Reply:
`combine_with_mut_non_residual_top` was bugged until a recent change, and was equivalent to the residual-top version. I fixed one incorrect usage in the benchmarks PR. I'm not surprised to see another one here

Reply:
Fixed.